src/ps_wait.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+const cli = require('heroku-cli-util')
+
+module.exports = function * (context, heroku, delay) {
+  const wait = require('co-wait')
+
+  if (context.flags['with-run'] && context.flags['type']) {
+    cli.exit(1, 'Cannot specify both --type and --with-run')
+  }
+
+  let releases = yield heroku.request({
+    path: `/apps/${context.app}/releases`,
+    partial: true,
+    headers: {
+      'Range': 'version ..; max=1, order=desc'
+    }
+  })
+
+  if (releases.length === 0) {
+    cli.exit(1, `App ${context.app} has no releases`)
+  }
+
+  const latestRelease = releases[0]
+  let released = true
+  let interval = parseFloat(context.flags['wait-interval'])
+  if (!interval || interval < 0) {
+    interval = 10
+  }
+
+  if (delay) {
+    yield wait(delay)
+  }
+
+  while (true) {
+    let dynos = yield heroku.get(`/apps/${context.app}/dynos`)
+    dynos = dynos.filter((dyno) => dyno.type !== 'release')
+      .filter((dyno) => context.flags['with-run'] || dyno.type !== 'run')
+      .filter((dyno) => !context.flags.type || dyno.type === context.flags.type)
+
+    let onLatest = dynos.filter((dyno) => {
+      return dyno.state === 'up' && dyno.release.id === latestRelease.id
```

Comment:
N.B.: One known issue is that if you cut a release while you wait, I never pick up the new release id, and so this will wait forever. I have never run into this; we may want to leave as a known issue, but just letting you know.

Reply:
could we fix that by putting the release fetching inside the while loop?

Reply:
Sure, yeah. It's twice as many API calls but maybe doesn't matter. The "smart" fix would be to check the release version number instead of the id, and if the release of any of the running dynos is *greater* than the one we're looking for, wait for that instead.